pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -0,0 +1,151 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, fetchurl
+, runCommand
+, patchutils
+, pkg-config
+, glibcLocales
+, cmake
+, python3
+, libpng
+, zlib
+, eigen
+, protobuf
+, nsync
+, howard-hinnant-date
+, re2
+, nlohmann_json
+, boost
+, oneDNN
+}:
+
+let
+  externals = {
+    pytorch_cpuinfo = fetchFromGitHub {
+      owner = "pytorch";
+      repo = "cpuinfo";
+      rev = "5916273f79a21551890fd3d56fc5375a78d1598d";
+      sha256 = "sha256-nXBnloVTuB+AVX59VDU/Wc+Dsx94o92YQuHp3jowx2A=";
+    };
+    onnx = fetchFromGitHub {
+      owner = "onnx";
+      repo = "onnx";
+      rev = "v1.12.0";
+      sha256 = "sha256-3awGaKbzvZraGFJWoKIfHDh7qm6gWWfiO3bpGTcMLr0=";
+    };
+    "SafeInt/safeint" = fetchFromGitHub {
+      owner = "dcleblanc";
+      repo = "SafeInt";
+      rev = "ff15c6ada150a5018c5ef2172401cb4529eac9c0";
+      sha256 = "sha256-PK1ce4C0uCR4TzLFg+elZdSk5DdPCRhhwT3LvEwWnPU=";
+    };
+    flatbuffers = fetchFromGitHub {
+      owner = "google";
+      repo = "flatbuffers";
+      rev = "v1.12.0";
+      sha256 = "sha256-L1B5Y/c897Jg9fGwT2J3+vaXsZ+lfXnskp8Gto1p/Tg=";
+    };
+    nsync = nsync.src;
+    re2 = re2.src;
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "onnxruntime";
+  version = "1.12.1";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "onnxruntime";
+    rev = "v${version}";
+    sha256 = "sha256-mMBY+SZMbhP/WdDhZRlApbDQd70xm44+OUwTI34iRr0=";
+  };
+
+  preConfigure = lib.concatStringsSep "\n" (lib.mapAttrsToList (name: path: ''
+    rmdir cmake/external/${name}
+    ln -s ${path} cmake/external/${name}
+  '') externals);
+
+  # TODO: build server, and move .so's to lib output
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    python3
+  ];
+
+  patches = [
+    (fetchpatch {
+      name = "system-dnnl.patch";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/system-dnnl.diff?h=python-onnxruntime&id=0185531906bda3a9aba93bbb0f3dcfeb0ae671ad";
+      sha256 = "sha256-58RBrQnAWNtc/1pmFs+PkZ6qCsL1LfMY3P0exMKzotA=";
+    })
+  ];
+
+  postPatch =
+    let
+      abseil = fetchurl {
```

Comment:
I think it'd be better to add this to the `let ... in stdenv.mkDerivation` at line 25.